testhelpers/docker_registry.go

```diff
@@ -103,6 +148,9 @@ func (r *DockerRegistry) Start(t *testing.T) {
 	// wrap registry handler with authentication handler, defaulting to read-only
 	r.authnHandler = ReadOnly(r.regHandler)
 	if r.username != "" {
+		if r.customHandler != nil && r.customPrivileges != nil {
+			r.regHandler = Custom(r.regHandler, r.customPrivileges)
```

Comment:
Do you mean to be setting `r.authnHandler` here?

Reply:
(and if so, should we break after this line)

Reply:
No,  actually I am wrapping the regular handler with my custom behavior like this:
![wrapping](https://user-images.githubusercontent.com/1181799/125680717-8ff90437-4c62-45ce-9d59-c5ccfd9c0abe.png)
The idea is to keep the basic authentication logic for the registry, then in the cases, I want to simulate the read/write behavior access I will do it in the custom and all other requests I do not care about will go to the normal registry logic. So, lines 151 and 152 are ok exactly like that.

Reply:
Ohhh thank you for this explanation! That makes sense. Maybe add a comment?

Reply:
```suggestion
			// wrap registry handler with custom handler
			r.regHandler = Custom(r.regHandler, r.customPrivileges)
```

Reply:
Added at the following [commit](https://github.com/buildpacks/imgutil/pull/125/commits/4261e2896afc176ea6a4967a4bd255f780340892)